lib/codec.js

```diff
@@ -18,6 +18,7 @@
   exports.encodeBytes = f.lift(doEncodeBytes, _.identity);
   exports.encodeString = f.lift(doEncodeString, _.identity);
   exports.encodeSignedInt = f.lift(doEncodeSignedInt, _.identity);
+  exports.encodeShort = f.lift(doEncodeShort, _.identity);
```

Comment:
@galderz thank you for the explanation. You are right - I have added a comment based on my findings in [ISPN-6407](https://issues.jboss.org/browse/ISPN-6407). I have seen that you have already added tests in your upcoming pull request for testing the jboss String conversion.